src/test/resources/rest-api-spec/test/rescore/20_rescoring.yml

```diff
@@ -94,16 +153,17 @@ setup:
               body: "hello"
 
   - length: { hits.hits: 10 }
-  - match:  { hits.hits.0._source: { "body": "hello hello hello hello hello hello hello hello hello hello", "gender": "m" } }
-  - match:  { hits.hits.1._source: { "body": "hello hello hello hello hello hello hello hello hello bye", "gender": "m"} }
-  - match:  { hits.hits.2._source: { "body": "hello hello hello hello hello hello hello hello bye bye", "gender": "m"} }
-  - match:  { hits.hits.3._source: { "body": "hello hello hello hello hello hello hello bye bye bye", "gender": "m"} }
-  - match:  { hits.hits.4._source: { "body": "hello hello hello hello hello hello bye bye bye bye", "gender": "m"} }
-  - match:  { hits.hits.5._source: { "body": "hello hello hello hello hello bye bye bye bye bye", "gender": "f"} }
-  - match:  { hits.hits.6._source: { "body": "hello hello hello hello bye bye bye bye bye bye", "gender": "f"} }
-  - match:  { hits.hits.7._source: { "body": "hello hello hello bye bye bye bye bye bye bye", "gender": "f"} }
-  - match:  { hits.hits.8._source: { "body": "hello hello bye bye bye bye bye bye bye bye", "gender": "f"} }
-  - match:  { hits.hits.9._source: { "body": "hello bye bye bye bye bye bye bye bye bye", "gender": "f"} }
+  - match:  { hits.hits.0._source: { "body": "hello hello hello hello hello hello hello hello hello hello hello hello hello hello hello hello hello hello hello hello", "gender": "m" } }
+  - match:  { hits.hits.1._source: { "body": "hello hello hello hello hello hello hello hello hello hello hello hello hello hello hello hello hello hello hello bye", "gender": "m"} }
+  - match:  { hits.hits.2._source: { "body": "hello hello hello hello hello hello hello hello hello hello hello hello hello hello hello hello hello hello bye bye", "gender": "m"} }
+  - match:  { hits.hits.3._source: { "body": "hello hello hello hello hello hello hello hello hello hello hello hello hello hello hello hello hello bye bye bye", "gender": "m"} }
+  - match:  { hits.hits.4._source: { "body": "hello hello hello hello hello hello hello hello hello hello hello hello hello hello hello hello bye bye bye bye", "gender": "m"} }
+  - match:  { hits.hits.5._source: { "body": "hello hello hello hello hello hello hello hello hello hello hello hello hello hello hello bye bye bye bye bye", "gender": "f"} }
```

Comment:
🔴 this result is incorrect, for this query, for the query without the rescore all results should return man for the first 10, and woman for the second 10. 

You can get this by having more hello than bye for the men than for the woman. Just like in the original dataset example for this test.

Happy to explain better via video, as this can be tricky.